lib/pbench/agent/tool_meister.py

```diff
@@ -1349,19 +1357,13 @@ def _send_directory(self, directory, uri, ctx):
         tar_file = parent_dir / f"{target_dir}.tar.xz"
 
         try:
-            tar_process = self._create_tar(directory, tar_file)
-            if tar_process.returncode != 0:
-                # Re-try creating tar ball by suppressing all the warnings
-                tar_process = self._create_tar(
-                    directory, tar_file, ignore_warning=["none"]
-                )
-                if tar_process.returncode != 0:
-                    # Tar ball creation failed even after suppressing all the warnings,
-                    # we will now proceed to create an empty tar ball
-                    self._create_tar(
-                        Path("/dev/null"), tar_file, ignore_warning=["none"]
-                    )
-                    failures += 1
+            if self._create_tar(directory, tar_file, retry=True).returncode != 0:
+                # Tar ball creation failed even after suppressing all the warnings,
+                # we will now proceed to create an empty tar ball.
+                # TODO: it'd be better to be able to skip the PUT entirely if the
+                #  tar fails and simply log a failure without TDS waiting forever
+                if self._create_tar(Path("/dev/null"), tar_file, retry=True) != 0:
+                    raise ToolMeisterError
```

Comment:
I still think that our fallback here should be an empty _file_, not an empty _`tar` archive_.  (I.e., we can create an empty file without having to run the `tar` command, and that saves us from needing a _fourth_ fallback position.)